agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -264,6 +266,10 @@ private CompletableResultCode internalSend(
                 instrumentationKey,
                 startTime,
                 () -> {
+                  if (isStatsbeat) {
+                    statsbeatErrorCount.set(
+                        0); // reset error count when non-consecutive errors occur
+                  }
```

Comment:
how about setting a flag like `atLeastOneStatsbeatSuccess=true` instead

i don't think we need to shutdown if we see 3 consecutive errors if it's been running successfully already

Reply:
this is pretty much doing the same thing. when success reset the error counter. if it's 3 consecutive errors, we want to shut it down. i don't get your point.

Reply:
thinking it a bit more, i might understand now.. 
saying 1st request is successful, 2nd - 4th were errors.. then in that case we don't want to shut it down since we know it's temporary. that's a valid point.

Reply:
but what about 4th statsbeat is a success, in that case we don't want to shut it down either. it might be temporary too. 
should i increase the counter to max 10?